06.wc/wc.rb

```diff
@@ -10,36 +10,37 @@ def main
   else
     total = { line: 0, word: 0, byte: 0, file: 'total' }
     ARGV.each do |file|
-      total[:line] += build_data_for_file(file)[:line].to_i
-      total[:word] += build_data_for_file(file)[:word].to_i
-      total[:byte] += build_data_for_file(file)[:byte].to_i
+      calculate_result = build_data_for_file(file)
+      total[:line] += calculate_result[:line]
+      total[:word] += calculate_result[:word]
+      total[:byte] += calculate_result[:byte]
       output(build_data_for_file(file), options)
     end
     output(total, options) if ARGV.size > 1
   end
 end
 
 def build_data_for_stdin
-  inputs = $stdin.readlines.join
-  get_calculate_result(inputs, '')
+  combined_filename = $stdin.readlines.join
+  build_data_for_text(combined_filename, '')
 end
 
 def build_data_for_file(file)
   str = File.read(file)
-  get_calculate_result(str, file)
+  build_data_for_text(str, file)
 end
 
-def get_calculate_result(str, file)
+def build_data_for_text(str, file)
   { line: count_line(str), word: count_words(str), byte: count_byte(str), file: file }
 end
 
 def output(data, option)
-  print adjust_space(data[:line]) if option['l'] || lack_option(option)
-  print adjust_space(data[:word]) if option['w'] || lack_option(option)
-  print adjust_space(data[:byte]) if option['c'] || lack_option(option)
+  no_option = lack_option(option)
```

Comment:
左辺と右辺の名前が微妙に違って気持ち悪いパターンですね（`person = find_user`みたいな形）
そもそもlack_optionは1回しか使われないのでメソッド化しない、という手もあります。
```suggestion
  no_option = !option['l'] && !option['w'] && !option['c']
```

Reply:
メソッド化をしない処理に変更しました！

Reply:
個人的には別にこれぐらいいいじゃん、って思いますが、これだからrubocopは・・・おっと失礼w
なるほど、じゃあ仕方ないですね。ただ、`no_option = (option == { 'l' => false, 'w' => false, 'c' => false })`もちょっと可読性が悪い気がします。

`option.values`でハッシュの値を全部集めて、それから`none?`メソッドで真の値が1件でもあるかどうかをチェックする方法もあります。

```

irb(main):001:0> a=[false, false, true]
=> [false, false, true]
irb(main):002:0> b=[false, false, false] 
=> [false, false, false]
irb(main):003:0> c=[]
=> []
irb(main):007:0> a.none?
=> false
irb(main):008:0> b.none?
=> true
irb(main):009:0> c.none?
=> true
```

これもややトリッキーなところはありますが、コードはシンプルになりそうです。

Reply:
僕もこの部分、なんか気持ち悪いな〜🤔と思っていました。
````
no_option = (option == { 'l' => false, 'w' => false, 'c' => false })
````
なるほど、`values`メソッドと`none?`メソッドを組み合わせることで、かなり見やすくなった気がします！
````
no_option = option.values.none?
````
どちらも僕の知らないメソッドだったので、勉強になります✍️